Architecture/TodoApp/android/core/src/main/java/com/novoda/todoapp/tasks/data/model/Tasks.java

```diff
@@ -53,9 +53,13 @@ public boolean isEmpty() {
         return internalMap().isEmpty();
     }
 
-    public boolean isMostRecentAction(SyncedData<Task> syncedData) {
-        SyncedData<Task> currentSyncData = internalMap().get(syncedData.data().id());
-        return currentSyncData == null || syncedData.lastSyncAction() >= currentSyncData.lastSyncAction();
+    public boolean hasNoTaskMoreRecentThan(SyncedData<Task> syncedData) {
+        return hasNoTaskMoreRecentThan(syncedData.data().id(), syncedData.lastSyncAction());
+    }
+
+    public boolean hasNoTaskMoreRecentThan(Id id, long lastSyncAction) {
```

Comment:
👍 fixed in https://github.com/novoda/spikes/pull/130/commits/27928b970290dc85b8eede7f23cda93d8c04ce83